plugins/inputs/redis/redis.go

```diff
@@ -302,7 +347,7 @@ func (r *Redis) connect() error {
 // Reads stats from all configured servers accumulates stats.
 // Returns one of the errors encountered while gather stats (if any).
 func (r *Redis) Gather(acc telegraf.Accumulator) error {
-	if !r.connected {
+	if !r.connected || r.NodeDiscovery {
```

Comment:
Hmm if we re-create the slice of clients during each gather, should we also be calling `client.Close()` on the existing clients first inside `connect()`?

Reply:
I can test this out.  I haven't seen any rise in client connectivity in my deployed tests, but this would add some certainty.